pennylane_lightning/src/algorithms/CMakeLists.txt

```diff
@@ -4,16 +4,9 @@ set(CMAKE_CXX_STANDARD 17)
 set(ALGORITHM_FILES AdjointDiff.hpp AdjointDiff.cpp CACHE INTERNAL "" FORCE)
 add_library(lightning_algorithms STATIC ${ALGORITHM_FILES})
 
-target_include_directories(lightning_algorithms PUBLIC ${CMAKE_CURRENT_SOURCE_DIR} )
-target_link_libraries(lightning_algorithms PRIVATE lightning_simulator lightning_utils OpenMP::OpenMP_CXX)
-
-find_package(OpenMP)
-if(OpenMP_FOUND)
-    target_link_libraries(lightning_algorithms PRIVATE OpenMP::OpenMP_CXX)
-endif()
+target_link_libraries(lightning_algorithms PRIVATE pennylane_lightning_compile_options
+                                                   pennylane_lightning_external_libs)
 
+target_include_directories(lightning_algorithms PUBLIC ${CMAKE_CURRENT_SOURCE_DIR} )
+target_link_libraries(lightning_algorithms PRIVATE lightning_simulator lightning_utils)
```

Comment:
Since the `lightning_algorithms` target can be compiled as a separate module, do you think we should keep the OpenMP target link step in here?

Reply:
I found that `lightning_algorithms` anyway cannot be compiled independently, as it depends on other sub-targets. Let's just keep this and fix it when we really use sub-targets independently.